src/farm/yielder.cairo

```diff
@@ -0,0 +1,196 @@
+// SPDX-License-Identifier: MIT
+
+%lang starknet
+
+// Starkware dependencies
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.uint256 import Uint256
+
+// Project dependencies
+from openzeppelin.access.ownable.library import Ownable
+
+// Local dependencies
+from src.farm.library import CarbonableFarmer
+
+//
+// Constructor
+//
+
+@constructor
+func constructor{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    carbonable_project_address: felt, owner: felt
+) {
+    // Desc:
+    //   Initialize the contract with the given parameters -
+    //   This constructor uses a dedicated initializer that mainly stores the inputs
+    // Implicit args:
+    //   syscall_ptr(felt*)
+    //   pedersen_ptr(HashBuiltin*)
+    //   range_check_ptr
+    // Explicit args:
+    //   carbonable_project_address(felt): Address of the corresponding Carbonable project
+    //   owner(felt): Owner address
+    // Returns:
+    //   None
+    CarbonableFarmer.initializer(carbonable_project_address);
+    Ownable.initializer(owner);
+    return ();
+}
+
+//
+// Getters
+//
+
+@view
+func carbonable_project_address{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    ) -> (carbonable_project_address: felt) {
+    // Desc:
+    //   Return the associated carbonable project
+    // Implicit args:
+    //   syscall_ptr(felt*)
+    //   pedersen_ptr(HashBuiltin*)
+    //   range_check_ptr
+    // Returns:
+    //   carbonable_project_address(felt): Address of the corresponding Carbonable project
+    return CarbonableFarmer.carbonable_project_address();
+}
+
+@view
+func is_locked{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+    status: felt
+) {
+    // Desc:
+    //   Return the locked status of deposites and withdrawals
+    // Implicit args:
+    //   syscall_ptr(felt*)
+    //   pedersen_ptr(HashBuiltin*)
+    //   range_check_ptr
+    // Returns:
+    //   status(felt): Locked status
```

Comment:
Should be specify it's return FALSE when is locked and TRUE when unlocked

Reply:
I think I'll just add a comment to say TRUE when lock, FALSE when unlock (which was the common sense I guess, but you would get it right if I did that in the first time) :sweat_smile: